app/models/project.rb

```diff
@@ -106,8 +106,8 @@ class Project < ActiveRecord::Base
   validates_length_of :homepage, maximum: 255
   validates_length_of :identifier, in: 1..IDENTIFIER_MAX_LENGTH
   # downcase letters, digits, dashes but not digits only
-  validates_format_of :identifier, with: /\A(?!\d+$)[a-z0-9\-_]*\z/,
-                                   if: -> (p) { p.identifier_changed? }
+  validates :identifier, format: { with: /\A[a-z][a-z0-9\-_]*/ },
```

Comment:
You should reintroduce the `\z`, because otherwise things like this would match:
````
valid-until-now but somwhere later the invalid stuff begins !"§$%&/
````